### PR TITLE
Doc: Updated eos_cli_config_gen contribution guide to use host1 as centric testing

### PR DESCRIPTION
**Pull request by MaheshGSLAB**
## Change Summary

Updated eos_cli_config_gen contribution guide to use host1 as centric testing

## Related Issue(s)

Fixes https://github.com/aristanetworks/avd/pull/5465#discussion_r2123072328

## Component(s) name

`arista.avd.eos_cli_config_gen`

## Proposed changes
Ensure `host1` is used for all comprehensive command tests. Utilize other hosts(`host2`, `host3`, etc.) for commands that aren't broadly compatible or to enhance test coverage.

## How to test
NA

### User Checklist

- N/A

### Repository Checklist

- [x] My code has been rebased from devel before I start
- [x] I have read the [**CONTRIBUTING**](https://avd.arista.com/stable/docs/contribution/overview.html) document.
- [x] My change requires a change to the documentation and documentation have been updated accordingly.
- [x] I have updated [molecule CI](https://github.com/aristanetworks/avd/tree/devel/ansible_collections/arista/avd/molecule) testing accordingly. (check the box if not applicable)

### CODE REVIEW

File: docs/contribution/authoring_eos_cli_config_gen.md [hunk 1]
```diff
@@ -96,6 +96,7 @@ Run `pre-commit run --all`, this will trigger recompiling the schemas and the te
      - Add the additional test cases to the next host directory, e.g., `host_vars/host2/<key_name>.yml`, `host_vars/host3/<key_name>.yml`, etc.
      - Ensure each file represents unique scenarios or configurations to validate different combinations.
    - Ensure that the new keys are thoroughly tested with various valid values and edge cases.
+   - Ensure `host1` is used for all comprehensive command tests. Utilize other hosts(`host2`, `host2`, etc.) for commands that aren't broadly compatible or to enhance test coverage.
```
Comment (laxmikantchintakindi) [inline docs/contribution/authoring_eos_cli_config_gen.md]:
```suggestion
   - Ensure `host1` is used for all comprehensive commands tests. Utilize other hosts(`host2`, `host3`, etc.) for commands that aren't broadly compatible or to enhance test coverage.
```

Comment (Vibhu-gslab) [inline docs/contribution/authoring_eos_cli_config_gen.md]:
```suggestion
   - Ensure `host1` is used for all comprehensive command tests. Use other hosts (e.g., `host2`, `host3`, etc.) for commands that aren't broadly compatible or to enhance test coverage.
```

Comment (MaheshGSLAB) [reply docs/contribution/authoring_eos_cli_config_gen.md]: updated with Vibhu's suggestion.

**Review by laxmikantchintakindi — APPROVED** [review]

**Review by Vibhu-gslab — APPROVED** [review]

File: docs/contribution/authoring_eos_cli_config_gen.md [hunk 2]
```diff
@@ -96,6 +96,7 @@ Run `pre-commit run --all`, this will trigger recompiling the schemas and the te
      - Add the additional test cases to the next host directory, e.g., `host_vars/host2/<key_name>.yml`, `host_vars/host3/<key_name>.yml`, etc.
      - Ensure each file represents unique scenarios or configurations to validate different combinations.
    - Ensure that the new keys are thoroughly tested with various valid values and edge cases.
+   - Ensure `host1` is used for all comprehensive command tests. Use other hosts (e.g., `host2`, `host3`, etc.) for commands that aren't broadly compatible or to enhance test coverage.
```
Comment (gmuloc) [inline docs/contribution/authoring_eos_cli_config_gen.md]: what does `comprehensive` mean here? I am not sure I understand

Comment (MaheshGSLAB) [reply docs/contribution/authoring_eos_cli_config_gen.md]: When something is comprehensive, it aims to cover every aspect or element of a topic or situation. This can include all possible combinations of factors, variables, or items.

Comment (gmuloc) [reply docs/contribution/authoring_eos_cli_config_gen.md]: it feels that it is contradictory - the only way to achieve comprehensive coverage for all combinations in most case will imply distributing the commands over multiple hosts (because some are incompatible like using different unit for the same key for instance). 

Are you trying to say "you should use in `host1` commands / versions which are the most commonly supported together?

Comment (MaheshGSLAB) [reply docs/contribution/authoring_eos_cli_config_gen.md]: yes use host1 for all possible supported commands and use rest of hosts for commands which are not supported with host1 command.

Comment (ClausHolbechArista) [reply docs/contribution/authoring_eos_cli_config_gen.md]:
```suggestion
   - Use `host1` for all inputs. Also use other hosts (e.g., `host2`, `host3`, etc.) when you have mutually exclusive inputs or wish to test different values for a single key.
```

**Review by ClausHolbechArista — APPROVED** [review]
